temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -228,16 +247,60 @@ private void handleEventsBatch(List<HistoryEvent> events, boolean hasNextEvent)
       }
     }
 
+    // Look ahead to infer protocol messages
+    if (EventType.EVENT_TYPE_WORKFLOW_TASK_STARTED.equals(events.get(0).getEventType())) {
+      if (this.isReplaying()) {
+        for (HistoryEvent event : events) {
+          if (event.hasWorkflowExecutionUpdateAcceptedEventAttributes()) {
```

Comment:
This loop could be merged with the above one which is already looking ahead for version markers